### ISSUE TITLE
virtual check is wrong

### ISSUE THREAD

**Issue by blissend** [device42/salt_to_device42_sync#6]
The line on https://github.com/device42/salt_to_device42_sync/blob/fc9d8962f9fc89d78fee1953fc07f89e89e8b1b1/saltexplore.py#L133 is just checking for presence of virtual with value. This is wrong as shown in https://github.com/saltstack/salt/blob/659ec118ab21d1922d14013df7277d9f28c97678/salt/grains/core.py#L504 it by default has presence and physical as value that is returned. Multiple version of salt do the same including the latest. If this is not correct let me know.

Given the above, a potential solution is to just check if virtual doesn't have physical as value.

NOTE: Are there any unit tests done with this or provided with this?

**Comment by blissend** [comment]
Sample fix https://github.com/device42/salt_to_device42_sync/pull/7 properly marks inserted devices as physical type and rest as virtual. Subtypes not properly marked and it doesn't update types properly but those are preexisting issues.

**Comment by blissend** [comment]
Added another fix for virtual subtypes being wrong too. Salt puts the `node['virtual']` value as type of virtual hosts (e.g. KVM/VMware/Hyper-v) but they can but not always have subtypes (chroot/docker/jail etc.). This script doesn't account for the value in `node['virtual']` being the type of virtual host but it does now in #7. If there is a virtual_subtype I defined that as the virtual type for device42 instead.
